### PR TITLE
Port yuzu-emu/yuzu#2545: "core/core_timing_util: Use std::chrono types for specifying time units"

### PR DESCRIPTION
**Pull request by FearlessTobi**
See yuzu-emu/yuzu#2545 for more details.

**Original description**:
Simplifies the interface and bases it off of the `std::chrono time` units. This way any other time unit can have their conversions handled by the standard library (either implicitly through regular cross-unit conversions, or explicitly with duration_cast, should it be necessary).

---
This change is [Reviewable](https://reviewable.io/reviews/citra-emu/citra/4828)

### CODE REVIEW

**Comment by wwylele** [comment]
Would it be better to rename these functions to unified overloads `TimeToCycles`?

File: src/core/hle/kernel/shared_page.cpp [hunk 1]
```diff
@@ -104,7 +104,8 @@ void Handler::UpdateTimeCallback(u64 userdata, int cycles_late) {
     ++shared_page.date_time_counter;
 
     // system time is updated hourly
-    timing.ScheduleEvent(msToCycles(60 * 60 * 1000) - cycles_late, update_time_event);
+    timing.ScheduleEvent(msToCycles(std::chrono::milliseconds{60 * 60 * 1000}) - cycles_late,
```
Comment (B3n30) [inline src/core/hle/kernel/shared_page.cpp]: std::chrono::hours{1}

File: src/core/hle/kernel/thread.cpp [hunk 2]
```diff
@@ -184,7 +184,7 @@ void Thread::WakeAfterDelay(s64 nanoseconds) {
     if (nanoseconds == -1)
         return;
 
-    thread_manager.kernel.timing.ScheduleEvent(nsToCycles(nanoseconds),
+    thread_manager.kernel.timing.ScheduleEvent(nsToCycles(std::chrono::nanoseconds{nanoseconds}),
```
Comment (B3n30) [inline src/core/hle/kernel/thread.cpp]: make nanoseconds a std::chrono:: nanoseconds?

File: src/core/hle/kernel/timer.cpp [hunk 3]
```diff
@@ -57,8 +57,8 @@ void Timer::Set(s64 initial, s64 interval) {
         // Immediately invoke the callback
         Signal(0);
     } else {
-        kernel.timing.ScheduleEvent(nsToCycles(initial), timer_manager.timer_callback_event_type,
-                                    callback_id);
+        kernel.timing.ScheduleEvent(nsToCycles(std::chrono::nanoseconds{initial}),
```
Comment (B3n30) [inline src/core/hle/kernel/timer.cpp]: make initial std::chrono::nanoseconds

File: src/core/hle/kernel/timer.cpp [hunk 4]
```diff
@@ -87,7 +87,8 @@ void Timer::Signal(s64 cycles_late) {
 
     if (interval_delay != 0) {
         // Reschedule the timer with the interval delay
-        kernel.timing.ScheduleEvent(nsToCycles(interval_delay) - cycles_late,
+        kernel.timing.ScheduleEvent(nsToCycles(std::chrono::nanoseconds{interval_delay}) -
```
Comment (B3n30) [inline src/core/hle/kernel/timer.cpp]: make interval_delay a std::chrono::nanoseconds

File: src/core/hle/service/ir/ir_rst.cpp [hunk 5]
```diff
@@ -106,7 +106,8 @@ void IR_RST::UpdateCallback(u64 userdata, s64 cycles_late) {
     update_event->Signal();
 
     // Reschedule recurrent event
-    system.CoreTiming().ScheduleEvent(msToCycles(update_period) - cycles_late, update_callback_id);
+    system.CoreTiming().ScheduleEvent(
+        msToCycles(std::chrono::milliseconds{update_period}) - cycles_late, update_callback_id);
```
Comment (B3n30) [inline src/core/hle/service/ir/ir_rst.cpp]: update_period can be a std::chrono::milliseconds

**Comment by wwylele** [comment]
As discussed in discord, aggressively templatizing the stuff and relying on duration_cast can cause integer overflow, so I am fine with the current code. B3N30's idea is still nice, though.

File: src/core/hle/kernel/thread.cpp (hunk 2 above)
Comment (FearlessTobi) [reply src/core/hle/kernel/thread.cpp]: This just leads to an even bigger messs.

File: src/core/hle/service/nwm/nwm_uds.cpp [hunk 6]
```diff
@@ -874,8 +874,9 @@ ResultCode NWM_UDS::BeginHostingNetwork(const u8* network_info_buffer,
     connection_status_event->Signal();
 
     // Start broadcasting the network, send a beacon frame every 102.4ms.
-    system.CoreTiming().ScheduleEvent(msToCycles(DefaultBeaconInterval * MillisecondsPerTU),
-                                      beacon_broadcast_event, 0);
+    const s64 cycles = msToCycles(
+        std::chrono::milliseconds{static_cast<long>(DefaultBeaconInterval * MillisecondsPerTU)});
```
Comment (zhaowenlan1779) [inline src/core/hle/service/nwm/nwm_uds.cpp]: This is less accurate than the original code I think, as the 102.4ms will be cast to 102ms.

**Comment by FearlessTobi** [comment]
Closing as I don't see this being finished anytime soon.
If anyone wants to pick this up, please do.

**Pull request closed without merging**